aip/general/0236.md

```diff
@@ -0,0 +1,321 @@
+  string state = 1;
+
+  // An identifying string common to all preview mode logs for the product.
+  // Searching logs for this string will isolate all logs pertaining to this
+  // preview.
+  string log_prefix = 2;
+
+  // The start time of this preview.
+  google.protobuf.Timestamp start_time = 3;
+
+  // The latest stop time of this preview.
+  google.protobuf.Timestamp stop_time = 4;
+}
+```
+- `PreviewMetadata` **must** have the fields defined in the proto above and the
+  value of `state` **must** be `ACTIVE` or `INACTIVE`.
+- When a preview is first started, `preview_metadata` **must** be empty. It is
+  present on experiments once the "preview" method is used.
+- All preview metadata fields **must** be output only.
+- The system must change `state` to and from `ACTIVE` and `INACTIVE` when the
+  experiment is started or stopped, which can only be done by the "preview" and
+  "stop" custom methods.
+- The first time the "preview" custom method is used, the system **must** create
+  `preview_metadata` and do the following: 
+  - It **must** set the `state` to `ACTIVE`
+  - It **must** set a `log_prefix` string, which is a predefined constant
+    hard coded by the system developers. The same value is used for all preview
+    logs for the given resource type. An example is "FirewallPolicyPreviewLog".
+  - It **must** populate the `start_time` with the current time. 
+- When the "stop" custom method is used, the system **must** do the following:
+  - It **must** set the `state` to `INACTIVE`
+  - It **must** populate the `stop_time` with the current time. 
+- `start_time` **must** be updated every time the status is changed to `ACTIVE`.
+
+### APIs
+
+#### create
+
+- Creating a new experiment for preview **must** support the following use
+cases:
+  - Preview a new policy.
+  - Preview an update to an already live policy.
+  - Preview a deletion of a current policy.
+- For the update and delete use cases, the `policy` field in the `experiment`
+  **must** have the full payload of the live policy copied into it, including
+  the name. 
+  - The user **must** update the fields under test to the new intended state to 
+    preview an update.
+  - The user **must** set the `rules` to empty to preview a delete.
+- Previewing a new policy **must** be supported. 
+  - If the system does not support a nested collection without a live policy, 
+    the user **must** create a top-level resource which has the empty rules in
+    order to represent a no-op.
+  - Experiments are added as children of the no-op policy and their effect is
+    previewed.
+- If the system supports multiple previews for this experiment type, calling
+  "create" more than once **must** create multiple experiment resources. These
+  are represented as new subresources until the experimental policy limit is
+  reached.
+
+#### update
+
+- The name inside `policy` **must not** change.
+- The system **may** set the `state` to INACTIVE if the `state` was ACTIVE at
```

Comment:
I don't understand this point, to be honest... but I think it would be good to make the behavior consistent across APIs. I would be fine with this being a must, if we want to do it anywhere.